ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activityCompositeAnswer.component.ts

```diff
@@ -84,14 +84,25 @@ export class ActivityCompositeAnswer implements OnChanges {
                     this.childQuestionBlocks = [];
                     this.addBlankRow();
                 } else {
-                    this.childQuestionBlocks = newAnswers.map((rowOfAnswers: any[]) => {
+                    const questionsRows: ActivityQuestionBlock<any>[][] = newAnswers.map((rowOfAnswers: ActivityQuestionBlock<any>[]) => {
                         // assuming order of answers same as order of questions here.
                         // And adding braces and a return statement here fixes a bug that should not be happening
                         // Breaks when running the compiled version of SDK  but OK when including library via symlink
                         // TODO investigate what is going on. Perhaps differences in TypeScript compiler target Javascript config?
-                        return rowOfAnswers.map((answerContainer: any, index: number) =>
+                        return rowOfAnswers.map((answerContainer: ActivityQuestionBlock<any>, index: number) =>
                             this.buildBlockForChildQuestion(newBlock.children[index], answerContainer, newBlock.shown));
                     });
+
+                    const blankRow: ActivityQuestionBlock<any>[] = this.block.children.map((questionBlock: ActivityQuestionBlock<any>) =>
+                        this.buildBlockForChildQuestion(questionBlock, null, this.block.shown));
+
+                    this.childQuestionBlocks = questionsRows.map((currentRow: ActivityQuestionBlock<any>[]) => {
+                        if (currentRow.length !== blankRow.length) {
+                            const missedQuestions = blankRow.slice(currentRow.length);
+                            currentRow.push(...missedQuestions);
+                        }
+                        return currentRow;
+                    });
```

Comment:
Please add some comments here. Not sure what you are shooting for here.

Reply:
OK. I see the problem you are dealing with now. Thanks for the comments.
Consider updating the newAnswers.map block to deal with the blank row. Might simplify things:
e.g.:
```typescript
   const questionsRows: ActivityQuestionBlock<any>[][] = newAnswers.map((rowOfAnswers: ActivityQuestionBlock<any>[]) => {
                        return this.block.children.map((questionBlock, index) => {
                            const answerValue = index < rowOfAnswers.length ? rowOfAnswers[index] : null;
                            return this.buildBlockForChildQuestion(questionBlock, answerValue , questionBlock.shown);
                        });
                    });

I don't understand the use case as well as you do, so do check if this indeed deals with this situation!

Reply:
I tried to use your implementation, but it works differently. Can we leave it as-is?